src/metpy/calc/kinematics.py

```diff
@@ -567,9 +567,18 @@ def frontogenesis(potential_temperature, u, v, dx=None, dy=None, x_dim=-1, y_dim
 
     # Compute the angle (beta) between the wind field and the gradient of potential temperature
     psi = 0.5 * np.arctan2(shrd, strd)
```

Comment:
If you're interested in eliminating all trig:

We have $\tan(2 \psi) =\frac{shrd}{strd}$, which implies $\cos(2\psi)=\frac{strd}{\sqrt{shrd^2+strd^2}}$.  We use `psi` next line to find $\cos(\psi)$ and $\sin(\psi)$.

We show later that $\cos(2\psi) = 1 - 2 \sin^2(\psi) = 2 \cos^2(\psi) - 1$.  Rearranging those identities gives
$\cos(\psi) = \sqrt{\frac{1 + \cos(2 \psi)}{2} }$
$\sin(\psi) = \sqrt{\frac{1 - \cos(2 \psi)}{2} } = \sqrt{1-\cos^2(\psi)}$
This lets you replace three trig functions and a multiplication with three or four multiplications, three additions, three square roots, and a division, and four trips through the data with eight.  I have no idea whether the eliminating the trig calls makes up for the extra trips through the data.

Reply:
I ran through that too and wasn't wild about all the square roots. I'm not passionate about trying it but if someone wants to see about performance/accuracy, I'd entertain it.